xbmc/GUIInfoManager.cpp

```diff
@@ -8422,7 +8422,9 @@ std::string CGUIInfoManager::GetPlaylistLabel(int item, int playlistid /* = PLAY
     }
   case PLAYLIST_POSITION:
     {
-      return StringUtils::Format("%i", g_playlistPlayer.GetCurrentSong() + 1);
+      int currentSong = g_playlistPlayer.GetCurrentSong();
+      if (currentSong > -1)
+        return StringUtils::Format("%i", currentSong + 1);
```

Comment:
@phil65 <code>break;</code> is missing here. if currentSong is <= -1 execution will fall through PLAYLIST_RANDOM case, which is imo not intended.

Should read:
<pre>
case PLAYLIST_POSITION:
   {
      int currentSong = g_playlistPlayer.GetCurrentSong();
      if (currentSong > -1)
        return StringUtils::Format("%i", currentSong + 1);
      break;
   }
</pre>